azure-quantum/azure/quantum/plugins/cirq/targets/ionq.py

```diff
@@ -0,0 +1,170 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+from typing import TYPE_CHECKING, Any, Dict, Union
+
+try:
+    import cirq
+    from cirq_ionq import Job as CirqIonqJob
+    from cirq_ionq.results import QPUResult, SimulatorResult
+except ImportError:
+    raise ImportError(
+        "Missing optional 'cirq_ionq' dependency. \
+To install run: pip install azure-quantum[cirq]")
+
+from azure.quantum.target import IonQ
+from azure.quantum.plugins.cirq.targets.target import Target as CirqTarget
+
+if TYPE_CHECKING:
+    from azure.quantum import Workspace
+    from azure.quantum.job import Job as AzureJob
+
+
+class _IonQClient:
+    """Thin wrapper around Workspace to support cirq_ionq.Job"""
+    def __init__(self, workspace: "Workspace"):
+        self._workspace = workspace
+    
+    @staticmethod
+    def _to_ionq_status(status: str):
+        from azure.quantum._client.models._quantum_client_enums import JobStatus
+        _STATUS_DICT = {
```

Comment:
isn't this more broadly needed? i'm particular concerned if we need to add a new status, we shouldn't need to update more than one file.

Reply:
It is only needed here because it is specific to the way the cirq_ionq.Job object handles the job states. This mapping currently happens on the service side so here we are simply mapping it back to the original IonQ status values.